addons/event/models/event_slot.py

```diff
@@ -0,0 +1,189 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import pytz
+from math import modf
+
+from datetime import datetime, time, timedelta
+
+from odoo import _, api, Command, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import format_date, format_time
+from odoo.tools.float_utils import float_round
+
+
+class EventSlotWeekday(models.Model):
+    _name = "event.slot.weekday"
+    _description = "Event Slot Weekday"
+    _order = "sequence"
+
+    name = fields.Char("Name", required=True)
+    sequence = fields.Integer("Sequence", required=True)
+    color = fields.Integer("Color")
+
+
+class EventSlot(models.Model):
+    _name = "event.slot"
+    _description = "Event Slot"
+    _order = "is_recurrent DESC, date, start_hour, end_hour, id"
+
+    name = fields.Char("Name", compute="_compute_name", store="True")
+    event_id = fields.Many2one("event.event", "Event", ondelete="cascade")
+    start_hour = fields.Float("Starting Hour", required=True, default=8.0)
+    end_hour = fields.Float("Ending Hour", required=True, default=12.0)
+    start_datetime = fields.Datetime("Start Datetimes", compute="_compute_datetimes")
+    end_datetime = fields.Datetime("End Datetimes", compute="_compute_datetimes")
+    is_recurrent = fields.Boolean(compute="_compute_is_recurrent", store=True)
```

Comment:
To me this field is not really necessary (especially if it's stored), seems you could directly check the `weekdays` or `date` field directly